test-sol/strategies/aaveV3/TestAaveV3Automation.t.sol

```diff
@@ -0,0 +1,352 @@
+        uint256 repayId = createAaveV3Repay();
+        uint256 repayFLId = createAaveV3FLRepay();
+
+        BundleBuilder bundleBuilder = new BundleBuilder();
+
+        uint64[] memory repayIds = new uint64[](2);
+        repayIds[INDEX_REPAY] = uint64(repayId);
+        repayIds[INDEX_REPAY_FL] = uint64(repayFLId);
+        repayBundleId = uint64(bundleBuilder.init(repayIds));
+    }
+
+    function _initBoostBundle() internal {
+        uint256 boostId = createAaveV3Boost();
+        uint256 boostFLId = createAaveV3FLBoost();
+
+        BundleBuilder bundleBuilder = new BundleBuilder();
+
+        uint64[] memory boostIds = new uint64[](2);
+        boostIds[INDEX_BOOST] = uint64(boostId);
+        boostIds[INDEX_BOOST_FL] = uint64(boostFLId);
+        boostBundleId = uint64(bundleBuilder.init(boostIds));
+    }
+
+    function _createAaveV3Position(bool _isSafe, address _wallet) internal {
+        gibTokens(_wallet, TokenAddresses.WETH_ADDR, INITIAL_TOKEN_AMOUNT);
+
+        user.supply(INITIAL_COLLATERAL_WETH_AMOUNT, _isSafe, collateralAsset.id, AAVE_MARKET);
+        user.borrow(_isSafe, AAVE_MARKET, INITIAL_DEBT_DAI_AMOUNT, 2, debtAsset.id);
+    }
+
+    function _subToAutomationBundles(bool _isSafe) internal {
+        AaveV3SubProxy subProxy = new AaveV3SubProxy(repayBundleId, boostBundleId);
+
+        AaveV3SubProxy.AaveSubData memory params = user.subToAutomationBundles(
+            _isSafe,
+            address(subProxy),
+            MIN_RATIO,
+            MAX_RATIO,
+            TARGET_RATIO_BOOST,
+            TARGET_RATIO_REPAY
+        );
+
+        repaySubId = SubStorage(SUB_STORAGE_ADDR).getSubsCount() - 2;
+        boostSubId = SubStorage(SUB_STORAGE_ADDR).getSubsCount() - 1;
+
+        repaySub = _formatRepaySub(params);
+        boostSub = _formatBoostSub(params);
+    }
+
+    function _walletSetUpBeforeEachTest(bool _isSafe) internal {
+        wallet = _isSafe ? user.safeAddr() : user.proxyAddr();
+        _createAaveV3Position(_isSafe, wallet);
+        _subToAutomationBundles(_isSafe);
+    }
+
+    //////////////////////////////// TESTs /////////////////////////////////////////////
+
+    function testAaveV3RepayStrategy() public {
+        _testAaveV3RepayStrategy(false);
+        _testAaveV3RepayStrategy(true);
+    }
+
+    function testAaveV3FLRepayStrategy() public {
+        _testAaveV3FLRepayStrategy(false);
+        _testAaveV3FLRepayStrategy(true);
+    }
+
+    function testAaveV3BoostStrategy() public {
+        _testAaveV3BoostStrategy(false);
+        _testAaveV3BoostStrategy(true);
+    }
+
+    function testAaveV3BoostFLStrategy() public {
+        _testAaveV3BoostFLStrategy(false);
+        _testAaveV3BoostFLStrategy(true);
+    }
+
+    function _testAaveV3RepayStrategy(bool _isSafe) internal {
+        _walletSetUpBeforeEachTest(_isSafe);
+
+        uint beforeRatio = trigger.getSafetyRatio(AAVE_MARKET, wallet);
+
+        (, uint256 borrowAmountBefore,,,, ) = pool.getUserAccountData(wallet);
```

Comment:
Since we have only one debt asset here, we could also read the balance directly from the variable debt token contract, like this:

`address variableDebtToken = debtAsset.variableDebtTokenAddress;`
`uint256 borrowAmountBefore = IERC20(variableDebtToken).balanceOf(wallet);`

This approach might be cleaner, as it avoids multiple commas. No need to change it—just suggesting an alternative method.